### ISSUE TITLE
[question] how to use qt6 package as feedstock

### ISSUE THREAD

**Issue by rockandsalt** [conan-io/conan-center-index#8249]
Hi,
I am trying to package a library (vtk) that makes pretty heavy use of the find_package feature in cmake so I set my generator to be "cmake_find_package".  Here's what [I have](https://github.com/rockandsalt/conan-center-index/blob/vtk_recipe/recipes/vtk/all/conanfile.py). 

But when I am building it, qt6 is unable to find the double-conversion package. I am not sure if it's a bug on the qt package side when using the "cmake_find_package" generator or if I didn't configure conan right. 
Regards

**Comment by ericLemanissier** [comment]
Please send us the full log of the build which fails, with the options you are using

**Comment by ericLemanissier** [comment]
I could reproduce a failure to find double-conversion with you vtk recipe: https://github.com/eirikb/proof-of-conan/runs/4371154123?check_suite_focus=true#step:8:744. cmake tries to use this file https://github.com/Kitware/VTK/blob/v9.1.0/CMake/Finddouble-conversion.cmake, whereas conan generated the proper file: https://github.com/eirikb/proof-of-conan/runs/4371154123?check_suite_focus=true#step:8:335
The solution is probably to remove all the `Findfoo.cmake` files in https://github.com/Kitware/VTK/blob/v9.1.0/CMake which are also generated by conan.

**Comment by rockandsalt** [comment]
No problem, thank you for the hint. I will try that.

**Comment by rockandsalt** [comment]
good news, I removed all the findFoo.cmake and it cleared this error. But I have another error, 
```
CMake Error at ../FindQt6.cmake:6 (message):
  Conan: Component 'Quick' NOT found in package 'Qt6'
```
the vtk qt integration requires qt quick. I am looking at the FindQt6.cmake generated by cmake and it's not there. Has it been renamed in conan. I am looking at the qt recipe and it seems to be there.

**Comment by ericLemanissier** [comment]
I'm working on this on https://github.com/ericLemanissier/conan-center-index/blob/patch-2/recipes/vtk/all/conanfile.py
I'll do a Pull request when it *starts* to work

**Comment by rockandsalt** [comment]
@ericLemanissier thanks for the help. How do you want to coordinate this? I still have a bit of time to work on this.

**Comment by ericLemanissier** [comment]
Well, you can merge my branch into yours. I fixed several issues already

**Comment by rockandsalt** [comment]
it seems conan doesn't have prebuilt binaries for the qt configuration we need. I built it once and it compiles on my computer. do you know how to upload the binaries to the remote?

**Comment by ericLemanissier** [comment]
You cannot upload to conan center. You have to use your own conan remote for this.
Regarding your vtk recipe, it looks like you will have to remove all the findfoo.cmake files from here too https://github.com/Kitware/VTK/blob/285daeedd58eb890cb90d6e907d822eea3d2d092/CMake/vtkInstallCMakePackage.cmake#L73

**Comment by rockandsalt** [comment]
yes, I noticed that too. but when I called `conan create . 9.1.0@user/channel --keep-source --build missing` with the fix I had. conan was rebuilding qt again which would take forever. how do I build qt and make it available for the rest of my package?

**Comment by ericLemanissier** [comment]
you have to rebuild qt each time one of qt's option, or qt's settings changes. Once it is done, the binary stays in your conan local cache, so you won't have to rebuild it again. The commit I did on your recipe changed qt's option, this is why you had to rebuild. You can set-up a remote to upload binaries, and be able to get them on several computers, but it won't change anything if all your build happen on the same computer, because the remote will be serving only binaries which already exist on said computer.

**Comment by rockandsalt** [comment]
yeah, I am not sure what is going on anymore. qt doesn't compile anymore. Throws an error about not finding freetype. Maybe it's something about my computer. I am looking at your proof-of-conan repo to see if I can run it in a clean environment. How do you run the github action? I see it needs 4 input.

**Comment by ericLemanissier** [comment]
Proof-of-conan is all @eirikb 's work: https://eirikb.github.io/proof-of-conan/

I cannot help you without a complete build log  of the failure

**Comment by eirikb** [comment]
@rockandsalt A bit OT, but if you wonder about the inputs to the action they should match one to one against the request your browser will make when you trigger a new build/run. In other words: Before clicking start hit F12 and look at the Network tab. Any questions you can e-mail me (or ping me con conan slack), since it's not related to this issue :)

**Comment by rockandsalt** [comment]
@ericLemanissier, I tested it on proof of conan. Yeah it was my computer, https://github.com/eirikb/proof-of-conan/runs/4396762333?check_suite_focus=true
